src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -53,27 +107,158 @@ void setGCalEventId(KCalCore::Incidence::Ptr event, const QString &id)
     event->setCustomProperty("jolla-sociald", "gcal-id", id);
 }
 
+QList<KDateTime> datetimesFromExRDateStr(const QString &exrdatestr, bool *isDateOnly)
+{
+    // possible forms:
+    // RDATE:19970714T123000Z
+    // RDATE;VALUE=DATE-TIME:19970714T123000Z
+    // RDATE;VALUE=DATE-TIME:19970714T123000Z,19970715T123000Z
+    // RDATE;TZID=America/New_York:19970714T083000
+    // RDATE;VALUE=PERIOD:19960403T020000Z/19960403T040000Z,19960404T010000Z/PT3H
+    // RDATE;VALUE=DATE:19970101,19970120
+
+    QList<KDateTime> retn;
+    QString str = exrdatestr;
+    *isDateOnly = false; // by default.
+
+    if (str.startsWith(QStringLiteral("exdate"), Qt::CaseInsensitive)) {
+        str.remove(0, 6);
+    } else if (str.startsWith(QStringLiteral("rdate"), Qt::CaseInsensitive)) {
+        str.remove(0, 5);
+    } else {
+        SOCIALD_LOG_ERROR("not an ex/rdate string:" << exrdatestr);
+        return retn;
+    }
+
+    if (str.startsWith(';')) {
+        str.remove(0,1);
+        if (str.startsWith("DATE-TIME:", Qt::CaseInsensitive)) {
+            str.remove(0, 10);
+            QStringList dts = str.split(',');
+            Q_FOREACH (const QString &dtstr, dts) {
+                if (dtstr.endsWith('Z')) {
+                    // UTC
+                    KDateTime kdt = KDateTime::fromString(dtstr, RFC5545_KDATETIME_FORMAT);
+                    kdt.setTimeSpec(KDateTime::Spec::UTC());
+                    retn.append(kdt);
+                } else {
+                    // Floating time
+                    KDateTime kdt = KDateTime::fromString(dtstr, RFC5545_KDATETIME_FORMAT_NTZC);
+                    kdt.setTimeSpec(KDateTime::Spec::ClockTime());
+                    retn.append(kdt);
+                }
+            }
+        } else if (str.startsWith("DATE:", Qt::CaseInsensitive)) {
+            str.remove(0, 5);
+            QStringList dts = str.split(',');
+            Q_FOREACH(const QString &dstr, dts) {
+                QDate date = QDate::fromString(dstr, RFC5545_QDATE_FORMAT);
+                KDateTime kdt(date, KDateTime::Spec::ClockTime());
+                retn.append(kdt);
+            }
+        } else if (str.startsWith("PERIOD:", Qt::CaseInsensitive)) {
+            SOCIALD_LOG_ERROR("unsupported parameter in ex/rdate string:" << exrdatestr);
+            // TODO: support PERIOD formats, or just switch to CalDAV for Google sync...
+        } else if (str.startsWith("TZID=")) {
+            QString tzidstr = str.mid(0, str.indexOf(':'));
+            KTimeZone tz = KSystemTimeZones::zone(tzidstr);
```

Comment:
Good question. Falling back to clock time and printing warning could be best. But at least this current method of allowing events with invalid timezone to be created might have strange consequences.